Mathlib/Analysis/Convex/Basic.lean

```diff
@@ -595,3 +601,44 @@ protected theorem starConvex (K : Submodule 𝕜 E) : StarConvex 𝕜 (0 : E) K
   K.convex K.zero_mem
 
 end Submodule
+
+section CommSemiring
+
+variable {R : Type*} [CommSemiring R]
+variable (A : Type*) [Semiring A] [Algebra R A]
+variable {M : Type*} [AddCommMonoid M] [Module A M] [Module R M] [IsScalarTower R A M]
+variable [PartialOrder R] [PartialOrder A]
+
+lemma convex_of_nonneg_algebraMap {s : Set M} (halg : ∀ ⦃r : R⦄, 0 ≤ r → 0 ≤ algebraMap R A r)
+    (hs : Convex A s) : Convex R s := by
```

Comment:
Again, is this meaningfully more general than `SMulPosMono`? We have [algebraMap_nonneg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Module/Algebra.html#algebraMap_nonneg) which you use below.